botocore/client.py

```diff
@@ -491,7 +492,16 @@ def _make_api_call(self, operation_name, api_params):
         )
 
         if http.status_code >= 300:
-            raise ClientError(parsed_response, operation_name)
+            if parsed_response.get("Error", {}).get("Code"):
+                error_code = parsed_response["Error"]["Code"]
+                if "." in error_code:
+                    error_code = error_code.replace(".", "")
+                if not error_code[0].isalpha():
+                    error_class = ClientError
```

Comment:
I still think that a ``404`` exception will be created with this logic. You may want to hoist it up to ``parsed_response...`` if statement.